### DISCUSSION TITLE
Use $WebEvent in Function Titles, Messages, etc.

### DISCUSSION THREAD

**Discussion by andoni-galardi** [Badgerati/Pode.Web#560] [Q&A]
Hi,

I am just starting with Pode and I have a basic question. Is it possible to use $WebEvent outside ScriptBlocks?

My first goal is to show the message: Welcome {user} in the home page after login, but I can't.

```
 Set-PodeWebHomePage -Layouts @(
        New-PodeWebHero -Title "Welcome $($WebEvent.Auth.User['name'])!" -Message "This is your home page!" -Content @(
            New-PodeWebText -Value "Here is some text!" -InParagraph -Alignment Center
        )
)
```

It raised a NullArray error: in $WebEvent.Auth.User['name']

I am almost sure I am missing something really basic, but I can't find any info or example.

**Comment by Clifra-Jones** [comment]
For any one reading this. The problem is that Set-PodeWebHomePage only supports Layouts! It doesn't have a ScriptBlock property so that the $webEvents variable is not exposed.
You would have to change yourt home page to use Add-PodeWebPage, but then I don't knw how that effects using templates.
